4_compile.py

```diff
@@ -169,7 +173,8 @@ def configure_optimizers(self):
     model = AutoModelForSequenceClassification.from_pretrained(
         "distilbert-base-uncased", num_labels=2
     )
-
+    
+    # First model inference will trigger compilation
     model = torch.compile(model)
```

Comment:
Passing in a sample batch first might help or ensuring the training run is long enough so compilation time is amortized

Reply:
Weird question, when I use DeepSpeed, I would have to do that on each device separately?

Reply:
Nice, I was actually able to shave of a minute by doing that. 

Now, the only problem is that the model doesn't work for inference when I load it from a checkpoint. Any ideas for how to fix that?

Reply:
So what I did was 

```python
    model.to(torch.device("cuda:0"))
    model = torch.compile(model)

    for batch_idx, batch in enumerate(train_loader):
        model.train()
        for s in ["input_ids", "attention_mask", "label"]:
            batch[s] = batch[s].to(torch.device("cuda:0"))
        break

    outputs = model(
        batch["input_ids"],
        attention_mask=batch["attention_mask"],
        labels=batch["label"],
    )

    lightning_model = LightningModel(model)
    # start timing and training below
```

This was for a single-GPU context. But how would I do that in a multi-GPU context where each GPU keeps its own model. Would I have to compile it on each device or is it okay to compile it on the first device? Any recommendations for DeepSpeed contexts?

Reply:
After training, is there any command required to make it compatible with inference mode perhaps so that I can evaluate the model on the test set?

Reply:
@msaroufim Can you elaborate on this comment?

> our suggestion would be not to use it anymore

Do you mean in general or in the context of `torch.compile`? Why would it not be suggested anymore?

Reply:
Only in the context of torch.compile